system/View/Filters.php

```diff
@@ -62,7 +62,7 @@ public static function date_modify($value, string $adjustment)
     /**
      * Returns the given default value if $value is empty or undefined.
      *
-     * @param mixed $value
+     * @param int|string|null $value
```

Comment:
This should support a lot more; `mixed` might have even been correct here.

Reply:
If it is really `mixed`, please use `object|resource|array|string|int|float|bool|null` for now.
Because there are too many `mixed` in the code base, and it is difficult to know it is really `mixed` or just not yet fixed.

Reply:
```php
    public function testDefaultManyValue()
    {
        $parser = new Parser($this->config, $this->viewsDir, $this->loader);

        $obj = (object) '';

        $data = [
            'value1' => 0.0,
            'value2' => [],
            'value3' => $obj->scalar,
        ];

        $template = '{ value1|default(foo) } { value2|default(bar) } { value3|default(baz) }';

        $parser->setData($data);
        $this->assertSame('foo bar baz', $parser->renderString($template));
    }
```
I already tests with above, and result error in array empty.

```diff
There was 1 failure:

1) CodeIgniter\View\ParserFilterTest::testDefaultManyValue
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'foo bar baz'
+'foo { value2|default(bar) } baz'

D:\Project\laragon\www\ci4\tests\system\View\ParserFilterTest.php:136
phpvfscomposer://D:\Project\laragon\www\ci4\vendor\phpunit\phpunit\phpunit:97

FAILURES!
Tests: 26, Assertions: 26, Failures: 1.
```

Reply:
That's odd! `empty([])` is definitely true - I wonder if this is somehow getting interpolated as `empty[''])` (which is false)? I think that we still allow `array` types here, and address this as a bug in another PR.

Reply:
Okay, already push doctype `default()`